### PR TITLE
DTSPB-1048 Solicitor grant of probate xui tests

### PR DESCRIPTION
**Pull request by mounikahmcts**
https://tools.hmcts.net/jira/browse/TA-325
https://tools.hmcts.net/jira/browse/DTSPB-1048

Added xui tests for 

1. Apply for probate single executor
2. Apply for probate multi executor
3. Apply for probate - will left annexed - solicitor main applicant
4. Apply for probate - will left annexed - solicitor not the main applicant
5. Apply for probate - no will

All the solicitor tests will run against XUI and caseworker tests will run against ccd ui

### CODE REVIEW

File: src/test/end-to-end/helpers/PuppeteerHelper.js [hunk 1]
```diff
@@ -13,18 +13,29 @@ class PuppeteerHelper extends Helper {
 
     async waitForNavigationToComplete(locator) {
         const page = this.helpers[helperName].page;
+        const promises = [];
 
-        const promises = [
-            page.waitForNavigation({timeout: 60000, waitUntil: ['domcontentloaded', 'networkidle0']}) // The promise resolves after navigation has finished
-        ];
+        if (!testConfig.TestForXUI) {
```
Comment (mounikahmcts) [inline src/test/end-to-end/helpers/PuppeteerHelper.js:18]: XUI is not able to resolve it.

File: src/test/end-to-end/helpers/PuppeteerHelper.js [hunk 2]
```diff
@@ -13,18 +13,29 @@ class PuppeteerHelper extends Helper {
...
         if (locator) {
             promises.push(page.click(locator));
         }
-
         await Promise.all(promises);
     }
 
+    async clickTab(tabTitle) {
+        const helper = this.helpers[helperName];
+        if (testConfig.TestForXUI) {
```
Comment (mounikahmcts) [inline src/test/end-to-end/helpers/PuppeteerHelper.js:30]: Angular mat-tab is used in XUI which will render dynamically and not through clickable link

File: src/test/end-to-end/pages/caseDetails/caseDetails.js [hunk 3]
```diff
@@ -24,17 +25,23 @@ module.exports = async function (caseRef, tabConfigFile, dataConfigFile, nextSte
...
         let eventSummaryPrefix = nextStep;
 
         eventSummaryPrefix = eventSummaryPrefix.replace(/\s+/g, '_').toLowerCase() + '_';
 
         await I.waitForText(nextStep, testConfig.TestTimeToWaitForText || 60);
         await I.waitForText(endState, testConfig.TestTimeToWaitForText || 60);
-        await I.waitForText(eventSummaryPrefix + dataConfigFile.summary, testConfig.TestTimeToWaitForText || 60);
-        await I.waitForText(eventSummaryPrefix + dataConfigFile.comment, testConfig.TestTimeToWaitForText || 60);
+
+        if (dataConfigKeys) {
```
Comment (mounikahmcts) [inline src/test/end-to-end/pages/caseDetails/caseDetails.js:41]: For some of the events we do not need to provide summary and comment

**Review by phaniperla — APPROVED** [review]
Approved

File: src/test/end-to-end/pages/caseDetails/solicitorApplyProbate/copiesTabConfig.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      "How many sealed and certified copies of the grant do you need for use outside the UK?",
+      "How do you want to pay?"
+    ],
+    "dataKeys": [
+      "page4_extraCopiesUK",
+      "page4_outsideUKGrantCopies",
+      "page5_paymentType"
+    ]
+  }
+
```
Comment (vinceholt) [inline src/test/end-to-end/pages/caseDetails/solicitorApplyProbate/copiesTabConfig.json:14]: very minor issue - if you add a carriage return at end of line - this complaining symbol will go. Most lint rules do requrie a carriage return at end of line

File: src/test/end-to-end/pages/newCase/newCase.js [hunk 5]
```diff
@@ -3,10 +3,12 @@
...
 
-module.exports = async function () {
+module.exports = function () {
 
     const I = this;
 
-    await I.waitForText(newCaseConfig.waitForText, testConfig.TestTimeToWaitForText);
-    await I.waitForNavigationToComplete(newCaseConfig.locator);
+    I.waitForText(newCaseConfig.waitForText, testConfig.TestTimeToWaitForText);
+
```
Comment (vinceholt) [inline src/test/end-to-end/pages/newCase/newCase.js:11]: these 2 need to be async / await to wait for promise to resolve, possibly linked to removal of  waitForNavigation?

File: src/test/end-to-end/helpers/PuppeteerHelper.js [hunk 6]
```diff
@@ -13,18 +13,29 @@ class PuppeteerHelper extends Helper {
 
     async waitForNavigationToComplete(locator) {
         const page = this.helpers[helperName].page;
+        const promises = [];
 
-        const promises = [
-            page.waitForNavigation({timeout: 60000, waitUntil: ['domcontentloaded', 'networkidle0']}) // The promise resolves after navigation has finished
-        ];
```
Comment (vinceholt) [inline src/test/end-to-end/helpers/PuppeteerHelper.js:19]: This is interesting - unsure as to why this wouldn't work for xui, both are webpages loaded in a web browser? May cause complications down the line e.g. newCase

**Comment by vinceholt** [comment]
Would suggest raising a separate JIRA ticket to investigate the 403 error which necessitated the code change to remove the waitForNavigation as this may well be an issue with EXUI code and we may have found a bug. We could then fix bug and get the waitForNavigation reinstated - just a thought!

File: src/test/end-to-end/helpers/PuppeteerHelper.js (hunk 6 above)
Comment (mounikahmcts) [reply src/test/end-to-end/helpers/PuppeteerHelper.js:19]: As discussed raised https://tools.hmcts.net/jira/browse/DTSPB-1049 for investigation

**Comment by mounikahmcts** [comment]
> Would suggest raising a separate JIRA ticket to investigate the 403 error which necessitated the code change to remove the waitForNavigation as this may well be an issue with EXUI code and we may have found a bug. We could then fix bug and get the waitForNavigation reinstated - just a thought!

https://tools.hmcts.net/jira/browse/DTSPB-1049

**Pull request merged**